### ISSUE TITLE
Stuck at scanning deprecations in offline mode

### ISSUE THREAD

**Issue by WarningImHack3r** [WarningImHack3r/npm-update-dependencies#110]
### What is the issue?

NPM commands 401-ing (incorrect or missing password), 1 non-default registry found (without trailing slash), stuck [here](https://github.com/WarningImHack3r/npm-update-dependencies/blob/b74fc25e6cf6a9df7cc5029890a153664dc9bde7/src/main/kotlin/com/github/warningimhack3r/npmupdatedependencies/backend/engine/NPMJSClient.kt#L40), 8/29 deps scanned

Logs:
- [x] Add debug "Starting batch n deprecations/updates" end .also
- [x] Warn if only error on shellrunner, print first 2 non-empty lines
- [x] Better log for deprecationbanner leaving if before scan
- [ ] More to see (including when issue fixed)

### What are the steps to reproduce this issue?

\-

### What were you expecting to happen?

not hang

### Paste any relevant logs, error output, etc.

```shell
024-06-07 13:44:25,100 [ 410523]   FINE - #com.github.warningimhack3r.npmupdatedependencies.ui.banner.DeprecationBanner - Leaving: no deprecations found
2024-06-07 13:44:27,119 [ 412542]   FINE - #com.github.warningimhack3r.npmupdatedependencies.ui.banner.DeprecationBanner - Leaving: no deprecations found
2024-06-07 13:44:27,433 [ 412856]   FINE - #com.github.warningimhack3r.npmupdatedependencies.ui.annotation.DeprecationAnnotator - Registries not scanned yet, scanning now
2024-06-07 13:44:27,434 [ 412857]   INFO - #com.github.warningimhack3r.npmupdatedependencies.ui.statusbar.StatusBarHelper - Updating widget
2024-06-07 13:44:27,434 [ 412857]   FINE - #com.github.warningimhack3r.npmupdatedependencies.ui.statusbar.StatusBarHelper - Updating widget for project "XXX"
2024-06-07 13:44:27,434 [ 412857]   FINE - #com.github.warningimhack3r.npmupdatedependencies.ui.statusbar.StatusBarHelper - Widget updated for project "XXX"
2024-06-07 13:44:27,435 [ 412858]   INFO - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.RegistriesScanner - Starting to scan registries
2024-06-07 13:44:27,436 [ 412859]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.ShellRunner - Executing "npm config ls"
2024-06-07 13:44:27,703 [ 413126]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.ShellRunner - Executed command "npm config ls" with output:
; "user" config from /home/XXX/.npmrc

//URL/:_authToken = (protected) 
email = "XXX@XXX.com" 
registry = "URL" 
strict-ssl = false 

; Run `npm config ls -l` to show all defaults.

2024-06-07 13:44:27,704 [ 413127]   INFO - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.RegistriesScanner - Found registries: [URL]
2024-06-07 13:44:27,704 [ 413127]   INFO - #com.github.warningimhack3r.npmupdatedependencies.ui.statusbar.StatusBarHelper - Updating widget
2024-06-07 13:44:27,704 [ 413127]   FINE - #com.github.warningimhack3r.npmupdatedependencies.ui.statusbar.StatusBarHelper - Updating widget for project "XXX"
2024-06-07 13:44:27,705 [ 413128]   FINE - #com.github.warningimhack3r.npmupdatedependencies.ui.statusbar.StatusBarHelper - Widget updated for project "XXX"
2024-06-07 13:44:27,705 [ 413128]   FINE - #com.github.warningimhack3r.npmupdatedependencies.ui.annotation.DeprecationAnnotator - Registries scanned
2024-06-07 13:44:27,706 [ 413129]   FINE - #com.github.warningimhack3r.npmupdatedependencies.ui.annotation.DeprecationAnnotator - Scanning for deprecations...
2024-06-07 13:44:27,707 [ 413130]   INFO - #com.github.warningimhack3r.npmupdatedependencies.ui.statusbar.StatusBarHelper - Updating widget
2024-06-07 13:44:27,707 [ 413130]   FINE - #com.github.warningimhack3r.npmupdatedependencies.ui.statusbar.StatusBarHelper - Updating widget for project "XXX"
2024-06-07 13:44:27,707 [ 413130]   FINE - #com.github.warningimhack3r.npmupdatedependencies.ui.statusbar.StatusBarHelper - Widget updated for project "XXX"
2024-06-07 13:44:27,710 [ 413133]   INFO - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.NPMJSClient - Getting deprecation status for package @jest/globals
2024-06-07 13:44:27,710 [ 413133]   INFO - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.NPMJSClient - Getting deprecation status for package @babel/preset-env
2024-06-07 13:44:27,710 [ 413133]   INFO - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.NPMJSClient - Getting deprecation status for package @sveltejs/adapter-static
2024-06-07 13:44:27,710 [ 413133]   INFO - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.NPMJSClient - Getting registry for package @sveltejs/adapter-static
2024-06-07 13:44:27,710 [ 413133]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.ShellRunner - Executing "npm v @sveltejs/adapter-static dist.tarball"
2024-06-07 13:44:27,710 [ 413133]   INFO - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.NPMJSClient - Getting deprecation status for package @types/eslint
2024-06-07 13:44:27,710 [ 413133]   INFO - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.NPMJSClient - Getting registry for package @types/eslint
2024-06-07 13:44:27,710 [ 413133]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.ShellRunner - Executing "npm v @types/eslint dist.tarball"
2024-06-07 13:44:27,710 [ 413133]   INFO - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.NPMJSClient - Getting deprecation status for package @sveltejs/vite-plugin-svelte
2024-06-07 13:44:27,712 [ 413135]   INFO - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.NPMJSClient - Getting registry for package @sveltejs/vite-plugin-svelte
2024-06-07 13:44:27,712 [ 413135]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.ShellRunner - Executing "npm v @sveltejs/vite-plugin-svelte dist.tarball"
2024-06-07 13:44:27,710 [ 413133]   INFO - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.NPMJSClient - Getting deprecation status for package vite
2024-06-07 13:44:27,713 [ 413136]   INFO - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.NPMJSClient - Getting registry for package vite
2024-06-07 13:44:27,713 [ 413136]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.ShellRunner - Executing "npm v vite dist.tarball"
2024-06-07 13:44:27,710 [ 413133]   INFO - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.NPMJSClient - Getting registry for package @babel/preset-env
2024-06-07 13:44:27,713 [ 413136]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.ShellRunner - Executing "npm v @babel/preset-env dist.tarball"
2024-06-07 13:44:27,710 [ 413133]   INFO - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.NPMJSClient - Getting deprecation status for package @sveltejs/kit
2024-06-07 13:44:27,710 [ 413133]   INFO - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.NPMJSClient - Getting deprecation status for package @babel/core
2024-06-07 13:44:27,715 [ 413138]   INFO - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.NPMJSClient - Getting registry for package @babel/core
2024-06-07 13:44:27,715 [ 413138]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.ShellRunner - Executing "npm v @babel/core dist.tarball"
2024-06-07 13:44:27,710 [ 413133]   INFO - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.NPMJSClient - Getting registry for package @jest/globals
2024-06-07 13:44:27,715 [ 413138]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.ShellRunner - Executing "npm v @jest/globals dist.tarball"
2024-06-07 13:44:27,715 [ 413138]   INFO - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.NPMJSClient - Getting registry for package @sveltejs/kit
2024-06-07 13:44:27,716 [ 413139]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.ShellRunner - Executing "npm v @sveltejs/kit dist.tarball"
2024-06-07 13:44:28,267 [ 413690]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.ShellRunner - Executed command "npm v @babel/preset-env dist.tarball" with output:

2024-06-07 13:44:28,268 [ 413691]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.NPMJSClient - No dist.tarball found for package @babel/preset-env, trying all registries
2024-06-07 13:44:28,289 [ 413712]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.ShellRunner - Executed command "npm v @types/eslint dist.tarball" with output:

2024-06-07 13:44:28,289 [ 413712]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.NPMJSClient - No dist.tarball found for package @types/eslint, trying all registries
2024-06-07 13:44:28,296 [ 413719]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.ShellRunner - Executed command "npm v vite dist.tarball" with output:

2024-06-07 13:44:28,296 [ 413719]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.NPMJSClient - No dist.tarball found for package vite, trying all registries
2024-06-07 13:44:28,298 [ 413721]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.ShellRunner - Executed command "npm v @babel/core dist.tarball" with output:

2024-06-07 13:44:28,298 [ 413721]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.NPMJSClient - No dist.tarball found for package @babel/core, trying all registries
2024-06-07 13:44:28,299 [ 413722]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.ShellRunner - Executed command "npm v @jest/globals dist.tarball" with output:

2024-06-07 13:44:28,299 [ 413722]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.NPMJSClient - No dist.tarball found for package @jest/globals, trying all registries
2024-06-07 13:44:28,301 [ 413724]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.ShellRunner - Executed command "npm v @sveltejs/adapter-static dist.tarball" with output:

2024-06-07 13:44:28,301 [ 413724]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.NPMJSClient - No dist.tarball found for package @sveltejs/adapter-static, trying all registries
2024-06-07 13:44:28,302 [ 413725]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.ShellRunner - Executed command "npm v @sveltejs/kit dist.tarball" with output:

2024-06-07 13:44:28,302 [ 413725]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.NPMJSClient - No dist.tarball found for package @sveltejs/kit, trying all registries
2024-06-07 13:44:28,365 [ 413788]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.ShellRunner - Executed command "npm v @sveltejs/vite-plugin-svelte dist.tarball" with output:

2024-06-07 13:44:28,366 [ 413789]   FINE - #com.github.warningimhack3r.npmupdatedependencies.backend.engine.NPMJSClient - No dist.tarball found for package @sveltejs/vite-plugin-svelte, trying all registries
```

### What OS are you experiencing the problem on?

Linux

### Duplicates

- [X] I have checked that my problem has not already been mentioned recently in an existing issue

**Comment by kitsunde** [comment]
Ah, so this is why my IDE was freezing. Any idea when this will end up on the marketplace?

**Comment by WarningImHack3r** [comment]
@kitsunde sorry to hear!
I’m actually waiting for https://github.com/semver4j/semver4j/issues/264 to be fixed so that #109 gets solved.

If you _really_ need this as soon as possible I can consider splitting those 2 fixes but I’d prefer waiting for a few days so I can ship an all in one bundle!

**Comment by kitsunde** [comment]
> @kitsunde sorry to hear! I’m actually waiting for [semver4j/semver4j#264](https://github.com/semver4j/semver4j/issues/264) to be fixed so that #109 gets solved.
> 
> If you _really_ need this as soon as possible I can consider splitting those 2 fixes but I’d prefer waiting for a few days so I can ship an all in one bundle!

All good man, no rush for my sake. Thanks for making a helpful plugin available. :)

**Comment by WarningImHack3r** [comment]
Here's an early build for that version as you wait: [npm-update-dependencies-2.3.3.zip](https://github.com/user-attachments/files/15794484/npm-update-dependencies-2.3.3.zip). I'll also try it tomorrow and see if the bug is fixed as expected; if not I'll push another fix.

Don't hesitate to open another issue if you still encounter the issue or another one! :)
